core/src/rigidbody.cpp

```diff
@@ -0,0 +1,289 @@
+}
+
+void RigidBody::create_capsule_rigid_dynamic(const float& radius,
+                                             const float& halfheight,
+                                             const float& mass,
+                                             const vec3& position,
+                                             const quat& rotation) {
+    if (m_static) {
+        return;
+    }
+    physx::PxShape* shape = NULL;
+    float p_x = position.x;
+    float p_y = position.y;
+    float p_z = position.z;
+    float r_w = rotation.w;
+    float r_x = rotation.x;
+    float r_y = rotation.y;
+    float r_z = rotation.z;
+    physx::PxVec3 p = physx::PxVec3(p_x, p_y, p_z);
+    physx::PxQuat q = physx::PxQuat(r_w, r_x, r_y, r_z);
+    if (m_material) {
+        shape = m_physics->createShape(physx::PxCapsuleGeometry(radius, halfheight), *m_material);
+    } else {
+        m_material = m_physics->createMaterial(0.3f, 0.3f, 0.6f);
+        shape = m_physics->createShape(physx::PxCapsuleGeometry(radius, halfheight), *m_material);
+    }
+    m_static = m_physics->createRigidStatic(PxTransform(p));
+    shape->setLocalPose(PxTransform(q));
+    m_static->attachShape(*shape);
+    m_scene->addActor(*m_static);
+    shape->release();
+}
+
+void RigidBody::create_capsule_rigid_static(const float& radius,
+                                            const float& halfheight,
+                                            const vec3& position,
+                                            const quat& rotation) {
+    if (m_dynamic) {
+        return;
+    }
+    physx::PxShape* shape = NULL;
+    float p_x = position.x;
+    float p_y = position.y;
+    float p_z = position.z;
+    float r_w = rotation.w;
+    float r_x = rotation.x;
+    float r_y = rotation.y;
+    float r_z = rotation.z;
+    physx::PxVec3 p = physx::PxVec3(p_x, p_y, p_z);
+    physx::PxQuat q = physx::PxQuat(r_w, r_x, r_y, r_z);
+    if (m_material) {
+        shape = m_physics->createShape(physx::PxCapsuleGeometry(radius, halfheight), *m_material);
+    } else {
+        m_material = m_physics->createMaterial(0.3f, 0.3f, 0.6f);
+        shape = m_physics->createShape(physx::PxCapsuleGeometry(radius, halfheight), *m_material);
+    }
+    m_static = m_physics->createRigidStatic(PxTransform(p));
+    shape->setLocalPose(PxTransform(q));
+    m_static->attachShape(*shape);
+    m_scene->addActor(*m_static);
+    shape->release();
+}
+
+void RigidBody::create_sphere_rigid_dynamic(const float& radius,
+                                            const float& mass,
+                                            const vec3& position,
+                                            const quat& rotation) {
+    if (m_static) {
+        return;
+    }
+    physx::PxShape* shape = NULL;
+    float p_x = position.x;
+    float p_y = position.y;
+    float p_z = position.z;
+    float r_w = rotation.w;
+    float r_x = rotation.x;
+    float r_y = rotation.y;
+    float r_z = rotation.z;
```

Comment:
does this data need to be set to a float p_x p_y .... can position.x ,position.y .... be directly passed into the functions PxVec3 / PxQuat instead like it is with half_size.x